libs/ledger-live-common/src/deviceSDK/commands/toggleOnboardingEarlyCheck.ts

```diff
@@ -0,0 +1,42 @@
+import Transport from "@ledgerhq/hw-transport";
+import { Observable } from "rxjs";
+
+export enum ToggleTypeP2 {
+  EnterChecking = 0x00,
+  ExitChecking = 0x01,
+}
+
+export type ToggleOnboardingEarlyCheckCmdEvent = { type: "success" };
+
+/**
+ * During the onboarding, makes the device enter or exit the early security check steps
+ *
+ * This command only puts (or moves out) the device to the state/step of the early security check.
+ * It does not starts any "security checks".
+ *
+ * This new “checking” APDU is only supported in WELCOME and WELCOME_STEP2 onboarding state.
+ * This command will throw an error if the device is not in one of those 2 onboarding state.
+ *
+ * @param transport a Transport instance
+ * @param p2 the APDU p2 value limited to ToggleTypeP2 enum values representing enter or exit
+ * @returns An observable that emits a success event. Otherwise throws an error.
+ */
+export function toggleOnboardingEarlyCheckCmd({
+  transport,
+  p2,
+}: {
+  transport: Transport;
+  p2: ToggleTypeP2;
+}): Observable<ToggleOnboardingEarlyCheckCmdEvent> {
+  return new Observable((subscriber) => {
+    transport
+      .send(0xe0, 0x03, 0x00, p2)
+      .then(() => {
```

Comment:
Doesn't the answer matter here? Are we sure that this promise will throw in case of an error and not just resolve with a specific error code or something like that?

Reply:
It's true that looking at it, we can't know that it would always fail/throw if the response is not `0x9000`

But there is a check in `send` on a list of status code `statusList`, and by default this list only contains `0x9000`: 

```TS
  send = async (
    cla: number,
    ins: number,
    p1: number,
    p2: number,
    data: Buffer = Buffer.alloc(0),
    statusList: Array<number> = [StatusCodes.OK]
  ): Promise<Buffer> => { }
```

So we're sure here in the `then` that it's a success.

Also the tests in `libs/ledger-live-common/src/deviceSDK/commands/toggleOnboardingEarlyCheck.test.ts` double check this ;)